### PR TITLE
update to MAME 240, iOS 13.4, remote target for macOS

### PR DESCRIPTION
**Pull request by ToddLa**
# update to MAME 240, iOS 13.4, remove target for macOS
* MAME 240
* update minimum version to iOS 13.4, tvOS 13.4, and macOS 10.15.4
* remove MAME4mac target, just build Catalyst from MAME4iOS target
* break the dam some more, added more swift code.

### CODE REVIEW

**Comment by eduo** [comment]
Will MAME for Mac be handled solely by the main MAME project from now on?

**Comment by ToddLa** [comment]
MAME4mac is now built from the iOS target, just build as Mac Catalyst

**Comment by yoshisuga** [comment]
That's cool that the iOS target could be used directly to build on Mac with Catalyst - I didn't know that was possible, or maybe that was always there?

We should fix the menus though later :) 

Screen Shot 2022-02-17 at 11 40 14 PM

**Comment by ToddLa** [comment]
> Nice changes! I kinda glossed over the file handling stuff but I'll run the build on my iPhone to see how it looks.

The file handling changes are some pre-work for a future feature, that should not effect current version. 

* create `software` and `shaders` folders, they currently are ignored. 

* read the list of file extensions we import from our own Info.plist, that way they are in one place. (next PR this list will grow....)

* add a version tag to our HASH.DAT database

* add new key to HASH database to lookup a single rom file (not in a software zip) this key not used (yet) but it's presence in the database won't effect current build.

**Comment by ToddLa** [comment]
> That's cool that the iOS target could be used directly to build on Mac with Catalyst - I didn't know that was possible, or maybe that was always there?

It was not possible until we bumped the minimum version to 13.0, I picked 13.4 cuz that is when proper keyboard support was added 

It took some playing around to get Xcode-build to do the right thing, I hope I did that right. 


> We should fix the menus though later :) 
> 
I have seen this before, I think it is a macOS bug, but I will investigate and fix....

File: xcode/MAME4iOS/OnscreenKeyboard/EmulatorKeyboard.swift [hunk 1]
```diff
@@ -68,6 +70,31 @@
 
  class EmulatorKeyboardView: UIView {
 
+     static var keyboardBackgroundColor = UIColor.systemGray6.withAlphaComponent(0.5)
```
[the remark this answers is not in the document]
Comment (ToddLa) [reply xcode/MAME4iOS/OnscreenKeyboard/EmulatorKeyboard.swift:73]: Yea in the view model, or the host/user can just set these "globals" before creating a Keyboard. 

The problem was that code like the Button/Key class needs access to some of these colors/constants and it does not have a context to get at the model. So a bigger change would be needed.

**Pull request merged**